### PR TITLE
added logs command

### PR DESCRIPTION
**Pull request by LivePersonFaas**

### CODE REVIEW

File: src/service/faas.service.ts [hunk 1]
```diff
@@ -354,6 +380,39 @@ export class FaasService implements IFaaSService {
...
+      end || Date.now()
+    }`;
+
+    if (levels && levels.length > 0) {
+      for (const level of levels) {
+        additionalParams += `&filterLevels=${level}`;
+      }
+    }
+
+    return this.getStream(
```
Comment (chrbrt) [inline src/service/faas.service.ts:407]: await the end of the stream so in the meantime a "Downloading lambda logs" hint can be displayed.

File: src/service/faas.service.ts [hunk 2]
```diff
@@ -354,6 +380,39 @@ export class FaasService implements IFaaSService {
...
+      {
+        urlPart,
+        additionalParams,
+      },
+      new LogsTransform(removeHeader),
```
Comment (chrbrt) [inline src/service/faas.service.ts:412]: Add a "Successfully downloaded lambda logs" message

**Review by chrbrt — CHANGES_REQUESTED** [review]

File: src/service/faas.service.ts [hunk 3]
```diff
@@ -354,6 +380,39 @@ export class FaasService implements IFaaSService {
...
+    removeHeader,
+  }: {
+    uuid: string;
+    start?: number;
+    end?: number;
+    levels?: string[];
+    removeHeader?: boolean;
+  }): Promise<void> {
+    const urlPart = '/logs/export';
+    let additionalParams = `&lambdaUUID=${uuid}&startTimestamp=${start}&endTimestamp=${
```
[the remark this answers is not in the document]
Comment (LivePersonFaas) [reply src/service/faas.service.ts:397]: Decision: won't be implemented

File: src/service/faas.service.ts (hunk 2 above)
Comment (LivePersonFaas) [reply src/service/faas.service.ts:412]: Decision: Won't be implemented (because of redirect to file)

File: src/service/faas.service.ts (hunk 1 above)
Comment (LivePersonFaas) [reply src/service/faas.service.ts:407]: Decision: Won't be implemented (because of redirect to file)

**Pull request merged**